src/backend/marsha/deposit/models.py

```diff
@@ -0,0 +1,211 @@
+"""
+Models for the deposit app of the Marsha project.
+
+In this base model, we activate generic behaviours that apply to all our models and enforce
+checks and validation that go further than what Django is doing.
+"""
+
+import logging
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from marsha.core.models import BaseModel, Playlist, UploadableFileMixin
+from marsha.core.utils.time_utils import to_timestamp
+
+
+logger = logging.getLogger(__name__)
+
+
+class FileDepository(BaseModel):
+    """Model representing a main container for a file depository."""
+
+    RESOURCE_NAME = "filedepositories"
+
+    playlist = models.ForeignKey(
+        to=Playlist,
+        related_name="%(class)ss",
+        verbose_name=_("playlist"),
+        help_text=_("playlist to which this file_depository belongs"),
+        # don't allow hard deleting a playlist if it still contains file_depository
+        on_delete=models.PROTECT,
+    )
+    lti_id = models.CharField(
+        max_length=255,
+        verbose_name=_("lti id"),
+        help_text=_("ID for synchronization with an external LTI tool"),
+        blank=True,
+        null=True,
+    )
+    title = models.CharField(
+        max_length=255,
+        verbose_name=_("title"),
+        help_text=_("title of the file_depository"),
+        null=True,
+        blank=True,
+    )
+    description = models.TextField(
+        verbose_name=_("description"),
+        help_text=_("description of the file_depository"),
+        null=True,
+        blank=True,
+    )
+    position = models.PositiveIntegerField(
+        verbose_name=_("position"),
+        help_text=_("position of this file_depository in the playlist"),
+        default=0,
+    )
+
+    class Meta:
+        """Options for the ``FileDepository`` model."""
+
+        db_table = "file_depository"
+        verbose_name = _("File depository")
+        verbose_name_plural = _("File depositories")
+        constraints = [
+            models.UniqueConstraint(
+                fields=["lti_id", "playlist"],
+                condition=models.Q(deleted=None),
+                name="file_depository_unique_idx",
+            )
+        ]
+
+    @staticmethod
+    def get_ready_clause():
+        """Clause used in lti.utils.get_or_create_resource to filter the file_depositories.
+
+        Only show file_depositories that have successfully been created.
+
+        Returns
+        -------
+        models.Q
+            A condition added to a QuerySet
+        """
+        return models.Q()
```

Comment:
I'm not sure I get it, "Only show file_depositories that have successfully been created" means a file depository is always ready ?

Reply:
You may ignore this comment, it a cookie cutter thing, so... ;)